psql2mysql.py

```diff
@@ -176,19 +182,27 @@ def do_prechecks(config):
     db.connect()
     tables = db.getSortedTables()
     for table in tables:
-        rows = db.scanTablefor4ByteUtf8Char(table)
-        if rows:
-            print("Table '%s' contains 4 Byte UTF8 characters, which are, "
+        incompatibles = db.scanTablefor4ByteUtf8Char(table)
+        if incompatibles:
+            print("Table '%s' contains 4 Byte UTF8 characters which are "
                   "incompatible with the 'utf8' encoding used by MariaDB" %
                   table.name
             )
-            print("The following rows are affected.")
-            x = PrettyTable()
-            x.field_names = ["Affected Column"] + list(table.columns)
-            for row in rows:
-                x.add_row([row[0]] + list(row[1]))
-            print x
-            raise Exception("UTF8 mb4 found")
+            print("The following rows are affected:")
+
+            output_table = PrettyTable()
+            output_table.field_names = [
+                "Primary Key",
```

Comment:
Key(s)

Reply:
Hm, sure? Isn't it always a single "Primary Key" made up of multiple columns? </hairsplitting mode=off> :-)

Reply:
Hm, true, that's the we we show it. I think it could be described both ways actually